### DISCUSSION TITLE
What is wrong with my command `sudo pikaur -S (sed "/#/D" gui_clients.txt | tr '\n' ' ')`?

### DISCUSSION THREAD

**Discussion by otakutyrant** [nushell/nushell#10911] [Q&A] [answered]
pikaur is a package manager for Arch Linux. `gui_clients.txt` includes a list of packages names, like this:

```
# To install all packages from this file, execute `pacman -S $(sed "/#/D" cli_clients.txt)`.

htop # Interactive Process Viewer

page-git # Pager

# Linux Manual
man-db
tld
...
```

However, when I execute `sudo pikaur -S (sed "/#/D" gui_clients.txt | tr '\n' ' ')`?, it complains:

```
:: warning: Following package cannot be found in AUR:
    arandr i3 i3status-rust xorg xorg-xinit xfce4-notifyd deepin-gtk-theme deepin-icon-theme    
```

And `sudo pikaur -S arandr i3 i3status-rust xorg xorg-xinit xfce4-notifyd deepin-gtk-theme deepin-icon-theme` works well.

**Comment by amtoine** [comment]
what does this give you?
```nushell
sed "/#/D" gui_clients.txt | tr '\n' ' '
```

**Reply by otakutyrant** [reply]
`                       arandr i3 i3status-rust xorg xorg-xinit xfce4-notifyd deepin-gtk-theme deepin-icon-theme ` as a pasted result of `sed "/#/D" gui_clients.txt | tr '\n' ' ' | xsel -b`.

[Here](https://github.com/otakutyrant/dotfiles/blob/master/gui_clients.txt) is the `gui_clients.txt`.

**Reply by amtoine** [reply]
ooooh i see

**Reply by amtoine** [reply]
try with
```nushell
sudo pikaur -S (sed "/#/D" gui_clients.txt | tr '\n' ' ' | str trim | split row " ")
```

**Reply by otakutyrant** [reply]
It works! Could you explain the mechanism please? It surprises me that the valid argument is a list of strings rather than a whole string.

**Reply by amtoine** [reply, marked as answer]
Nushell is a typed language and on top of all does not manipulate strings only as Bash does.

so when you try to pass the whole list of packages as a single string, the external command receives a single name, which is incorrect.

the syntax is either
```nushell
^external foo bar baz
```
or
```nushell
^external [foo, bar, baz]
```

> **Note**
> yours above is more like
> ```nushell
> ^external "foo bar baz"
> ```

> **Note**
> the last one with a list allows to put a long list of arguments on multiple lines
> ```nushell
> ^external [
>     --opt1
>     --opt2
>     foo
>     bar
>     baz
> ]
> ```

**Reply by otakutyrant** [reply]
I thank you.
